File: src (cpp)/animal/aves/peacock.cpp
#include "peacock.h"

Peacock::Peacock(int _x, int _y, int _weight) : def_ID(20),
                                                def_ratio_meat(20),
                                                def_ratio_plant(40) {
  ID = def_ID;
  position.SetX(_x);
  position.SetY(_y);
  ratio_meat = def_ratio_meat;
  ratio_plant = def_ratio_plant;
  is_land_animal = true;
  is_water_animal = false;
  is_air_animal = false;
  weight = _weight;
}
void Peacock::Interact() {
  std::cout << "The peacock is showing off its beautiful feather" << std::endl;
}

File: src/virtualzoo/animal/aves/Peacock.java
package virtualzoo.animal.aves;

import virtualzoo.animal.behavior.BehaviorTame;
import virtualzoo.animal.diet.Omnivora;

/** <p>
 * Kelas Peacock mendefinisikan atribut untuk peacock.
 * </p>
 * @author Holy Lovenia - 13515113
 * @version 3.0
 * @since 3.0
 */
public class Peacock extends Aves {

  /** <p>
   * Constructor
   *
   * Menciptakan peacock dengan posisi (locX,locY) dan berat animalWeight.
   * </p>
   * @param locX Nilai absis posisi.
   * @param locY Nilai ordinat posisi.
   * @param animalWeight Nilai berat peacock.
   */
  public Peacock(int locX, int locY, int animalWeight) {
    super();
    animalBehavior = new BehaviorTame();
    animalDiet = new Omnivora(animalWeight, 20, 40);
    id = 19;
    position.setLocation(locX, locY);
    isLandAnimal = true;
    isWaterAnimal = false;
    isAirAnimal = false;
  }

  /** <p>
   * Mengatur interaksi peacock.
   * </p>
   */
  public void interact() {
    System.out.println("The peacock is showing off its beautiful feather");
  }
}